java/client/src/test/java/glide/managers/CommandManagerTest.java

```diff
@@ -221,14 +223,38 @@ public void prepare_request_with_slot_key_routes(SlotType slotType) {
                 () -> assertFalse(requestBuilder.getRoute().hasSlotIdRoute()));
     }
 
+    @Test
+    public void prepare_request_with_by_address_route() {
+        CompletableFuture<Response> future = new CompletableFuture<>();
+        when(channelHandler.write(any(), anyBoolean())).thenReturn(future);
+        when(channelHandler.isClosed()).thenReturn(false);
+
+        ArgumentCaptor<RedisRequest.Builder> captor =
+                ArgumentCaptor.forClass(RedisRequest.Builder.class);
+
+        service.submitNewCommand(
+                CustomCommand, new String[0], new ByAddressRoute("testhost", 6379), r -> null);
+        verify(channelHandler).write(captor.capture(), anyBoolean());
+        var requestBuilder = captor.getValue();
+
+        assertAll(
+                () -> assertTrue(requestBuilder.hasRoute()),
+                () -> assertTrue(requestBuilder.getRoute().hasByAddressRoute()),
+                () -> assertEquals("testhost", requestBuilder.getRoute().getByAddressRoute().getHost()),
+                () -> assertEquals(6379, requestBuilder.getRoute().getByAddressRoute().getPort()),
+                () -> assertFalse(requestBuilder.getRoute().hasSimpleRoutes()),
+                () -> assertFalse(requestBuilder.getRoute().hasSlotIdRoute()),
+                () -> assertFalse(requestBuilder.getRoute().hasSlotKeyRoute()));
+    }
+
     @Test
     public void prepare_request_with_unknown_route_type() {
         CompletableFuture<Response> future = new CompletableFuture<>();
         when(channelHandler.write(any(), anyBoolean())).thenReturn(future);
 
         var exception =
                 assertThrows(
-                        IllegalArgumentException.class,
```

Comment:
If this is being changed to align with other existing clients, then I'm okay with this.